internal/cni-server/server.go

```diff
@@ -114,7 +139,156 @@ func (s *server) Start() error {
 	return nil
 }
 
+func (s *server) PutFd(ld net.Listener, unixconn net.Conn) {
+	tcpln := ld.(*net.TCPListener)
+	f, err := tcpln.File()
+	if err != nil {
+		log.Errorf("get tcp listen file err: %v", err)
+		return
+	}
+	inode, err := getInoFromFd(f)
+	if err != nil {
+		log.Errorf("get inode err: %v", err)
+		return
+	}
+	if err != nil {
+		log.Errorf("parse listen err: %v", err)
+	}
+	err = passfd.Put(unixconn.(*net.UnixConn), f)
+	if err != nil {
+		log.Errorf("passfd put fd err: %v", err)
+	}
+	f.Close()
+	delete(s.listeners, inode)
+}
+
+func (s *server) transferFd(ln net.Listener) {
```

Comment:
This method seems to have two roles, one is to transfer all fds from Server and the other is to transfer only `ln`. 
I suggest that it is more appropriate to split it into two methods for more readability.